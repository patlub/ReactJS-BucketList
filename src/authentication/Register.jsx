// styles
import '../App.css';

import React, {Component} from 'react';
import {Link, Redirect} from 'react-router-dom';

// configs
import {baseURL} from '../configs/config';

// Third party
import axios from 'axios'

class Register extends Component {
    constructor(props) {
        super(props);
        this.state = {
            loggedIn: localStorage.getItem('token'),
            name: '',
            email: '',
            password: '',
        };
    }

    /*
    * Fired when input changes
    * @param (event) event when input changes
    * */
    onInputChanged = (event) => {
        this.setState({
            [event.target.name]: event.target.value
        })
    };

    /**
     * This method handles user registration
     * @param {event} event when triggering register action.
     */
    onRegisterClick = (event) => {
        console.log(this.state);
        event.preventDefault();
        axios.post(`${baseURL}/auth/register`,
            {
                name: this.state.name,
                email: this.state.email,
                password: this.state.password
            })
            .then(function (response) {
                if (response.status === 201) {
                    localStorage.setItem('token', response.data.id);
                    this.setState({loggedIn: response.data.id});
                }
            }.bind(this))
            .catch(function (error) {
                console.log(error);
            });
    };

    render() {
        if (this.state.loggedIn) {
            return <Redirect to="/buckets"/>
        }

        return (
            <div className="container-fluid text-center ">
                <div className="col-md-4 col-md-offset-4 col-sm-9 auth-box">
                    <div className="panel panel-primary">
                        <div className="panel-heading">
                            <h4 className="panel-title">Sign up</h4>
                        </div>
                        <div className="panel-body">
                            <form onSubmit={this.onRegisterClick}>
                                <div className="modal-body">
                                    <div className="form-group">
                                        <input
                                            className="form-control"
                                            type="text"
                                            name="name"
                                            value={this.state.name}
                                            onChange={this.onInputChanged}
                                            placeholder="Name"
                                            required
                                        />
                                    </div>
                                    <div className="form-group">
                                        <input
                                            className="form-control"
                                            type="email"
                                            name="email"
                                            value={this.state.email}
                                            onChange={this.onInputChanged}
                                            placeholder="Email"
                                            required/>
                                    </div>
                                    <div className="form-group">
                                        <input
                                            className="form-control"
                                            type="password"
                                            name="password"
                                            value={this.state.password}
                                            onChange={this.onInputChanged}
                                            placeholder="Password"
                                            required/>
                                    </div>
                                </div>
                                <div className="modal-footer">
                                    <div>
                                        <button type="submit" className="btn btn-primary btn-lg btn-block">Register
                                        </button>
                                    </div>
                                    <div>
                                        <Link to="login" className="btn btn-link">Login</Link>
                                        <Link to="forgot_password" className="btn btn-link">Forgot Password</Link>
                                    </div>
                                </div>
                            </form>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}


export default Register;
