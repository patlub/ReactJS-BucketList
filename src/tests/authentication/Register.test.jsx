import React from 'react';
import { mount, shallow } from 'enzyme';
import ReactTestUtils from 'react-dom/test-utils'; // ES6
import { StaticRouter } from 'react-router';
import renderer from 'react-test-renderer';
import Register from '../../authentication/Register';


global.localStorage = {
  getItem: () => {},
};

describe('Component: Register', () => {
  it('Displays Register', () => {
    const rendered = renderer.create(
      <StaticRouter location="register" context={''}>
        <Register />
      </StaticRouter>,
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('updates state on input change', () => {
    const routerComponent = mount(
      <StaticRouter location="register" context={''}>
        <Register />
      </StaticRouter>,
    );
    const registerComponent = routerComponent.find('Register');
    const nameInput = registerComponent.find('[type="text"]');
    const emailInput = registerComponent.find('[type="email"]');
    const passwordInput = registerComponent.find('[type="password"]');

    ReactTestUtils.Simulate.change(nameInput.getDOMNode(), { target: { name: 'name', value: 'patrick' } });
    ReactTestUtils.Simulate.change(emailInput.getDOMNode(), { target: { name: 'email', value: 'patrick@gmail.com' } });
    ReactTestUtils.Simulate.change(passwordInput.getDOMNode(), { target: { name: 'password', value: 'secret' } });

    expect(routerComponent.find('Register').nodes[0].state.name).toEqual('patrick');
    expect(routerComponent.find('Register').nodes[0].state.email).toEqual('patrick@gmail.com');
    expect(routerComponent.find('Register').nodes[0].state.password).toEqual('secret');
  });
});

