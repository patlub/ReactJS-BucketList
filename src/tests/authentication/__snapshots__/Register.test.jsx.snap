// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Register Displays Register 1`] = `
<div
  className="container-fluid text-center "
>
  <div
    className="col-md-4 col-md-offset-4 col-sm-9 auth-box"
  >
    <div
      className="panel panel-primary"
    >
      <div
        className="panel-heading"
      >
        <h4
          className="panel-title"
        >
          Sign up
        </h4>
      </div>
      <div
        className="panel-body"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="modal-body"
          >
            <div
              className="form-group"
            >
              <input
                className="form-control"
                name="name"
                onChange={[Function]}
                placeholder="Name"
                required={true}
                type="text"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                required={true}
                type="email"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                required={true}
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            className="modal-footer"
          >
            <div>
              <button
                className="btn btn-primary btn-lg btn-block"
                type="submit"
              >
                Register
              </button>
            </div>
            <div>
              <a
                className="btn btn-link"
                href="/login"
                onClick={[Function]}
              >
                Login
              </a>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
  <div
    className="notification-container notification-container-empty"
  >
    <span />
  </div>
</div>
`;
